torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,696 @@
+        context_out_channels = [context_outs[i].shape[1] - hidden_dims[i] for i in range(len(context_outs))]
+        hidden_states: List[Tensor] = []
+        contexts: List[List[Tensor]] = []
+        for i, context_conv in enumerate(self.context_convs):
+            # As in the original paper, the actual output of the context encoder is split in 2 parts:
+            # - one part is used to initialize the hidden state of the recurent units of the update block
+            # - the rest is the "actual" context.
+            hidden_state, context = torch.split(context_outs[i], [hidden_dims[i], context_out_channels[i]], dim=1)
+            hidden_states.append(torch.tanh(hidden_state))
+            contexts.append(
+                torch.split(context_conv(F.relu(context)), [hidden_dims[i], hidden_dims[i], hidden_dims[i]], dim=1)
+            )
+
+        _, Cf, Hf, Wf = fmap1.shape
+        coords0 = make_coords_grid(batch_size, Hf, Wf).to(fmap1.device)
+        coords1 = make_coords_grid(batch_size, Hf, Wf).to(fmap1.device)
+
+        depth_predictions = []
+        for _ in range(num_iters):
+            coords1 = coords1.detach()  # Don't backpropagate gradients through this branch, see paper
+            corr_features = self.corr_block(centroids_coords=coords1, corr_pyramid=corr_pyramid)
+
+            depth = coords1 - coords0
+            if self.slow_fast:
+                # Using slow_fast GRU (see paper section 3.4). The lower resolution are processed more often
+                for i in range(1, self.num_level):
+                    # We only processed the smallest i levels
+                    level_processed = [False] * (self.num_level - i) + [True] * i
+                    hidden_states = self.update_block(
+                        hidden_states, contexts, corr_features, depth, level_processed=level_processed
+                    )
+            hidden_states = self.update_block(
+                hidden_states, contexts, corr_features, depth, level_processed=[True] * self.num_level
+            )
+            # Take the largest hidden_state to get the depth
+            hidden_state = hidden_states[0]
+            delta_depth = self.depth_head(hidden_state)
+            # in stereo mode, project depth onto epipolar
+            delta_depth[:, 1] = 0.0
+
+            coords1 = coords1 + delta_depth
+            up_mask = None if self.mask_predictor is None else self.mask_predictor(hidden_state)
+            upsampled_depth = upsample_flow((coords1 - coords0), up_mask=up_mask, factor=self.base_downsampling_ratio)
+            depth_predictions.append(upsampled_depth[:, :1])
+
+        return depth_predictions
+
+
+def _raft_stereo(
+    *,
+    weights: Optional[WeightsEnum] = None,
+    progress: bool = False,
+    shared_encoder_weight: bool = False,
+    # Feature encoder
+    feature_encoder_layers: Tuple[int, int, int, int, int],
+    feature_encoder_strides: Tuple[int, int, int, int],
```

Comment:
Are you sure these shouldn't be `List[int]`? Is it possible that someone might want to make the encoder deeper?

Reply:
Yes, as of now the number of layer on the encoder is not variable size. On the raft implementation we put check on this: https://github.com/pytorch/vision/blob/main/torchvision/models/optical_flow/raft.py#L122